rules/aip0131/method_rules_test.go

```diff
@@ -92,3 +96,158 @@ func TestResponseMessageName(t *testing.T) {
 		})
 	}
 }
+
+func TestHttpVerb(t *testing.T) {
+	// Set up testing permutations.
+	tests := []struct {
+		testName   string
+		verb       string
+		methodName string
+		msg        string
+	}{
+		{"Valid", "get", "GetBook", ""},
+		{"Invalid", "post", "GetBook", "HTTP GET"},
+		{"Irrelevant", "post", "AcquireBook", ""},
+	}
+
+	// Run each test.
+	for _, test := range tests {
+		t.Run(test.testName, func(t *testing.T) {
+			// Create the appropriate pattern based on the desired verb.
+			var httpRule *annotations.HttpRule
+			if test.verb == "post" {
```

Comment:
Can you do the set up outside of tests (for loop)? In general, a testing loop does running and comparing only. And from the `tests` cases, one should have a general idea of what are testing without digging into the testing loop.

Reply:
Done.